daemon/networkdriver/portallocator/portallocator.go

```diff
@@ -59,6 +68,29 @@ func NewErrPortAlreadyAllocated(ip string, port int) ErrPortAlreadyAllocated {
 	}
 }
 
+func init() {
+	const param = "/proc/sys/net/ipv4/ip_local_port_range"
+
+	if line, err := ioutil.ReadFile(param); err != nil {
+		log.Errorf("Failed to read %s kernel parameter: %s", param, err.Error())
+	} else {
+		var start, end int
+		if n, err := fmt.Fscanf(strings.NewReader(string(line)), "%d\t%d", &start, &end); n != 2 || err != nil {
```

Comment:
This approach does not work. There's no problem with parsing regular file, but somehow this kernel parameter does not like to be read byte after byte which apparently `fmt.Fscanf` does. Extract from `strace` output:

    open("/proc/sys/net/ipv4/ip_local_port_range", O_RDONLY|O_CLOEXEC) = 3
    futex(0x560290, FUTEX_WAIT, 0, NULL)    = 0
    read(3, "3", 1)                         = 1
    read(3, "", 1)                          = 0
    write(1, "Failed to parse port range from "..., 76Failed to parse port range from /proc/sys/net/ipv4/ip_local_port_range: EOF
    ) = 76
    close(3)                                = 0

But perhaps I'm doing it wrong... Here's my reproducer:

    package main
    
    import (
    	"flag"
    	"fmt"
    	"os"
    )
    
    func main() {
    	var filename = flag.String("file", "/proc/sys/net/ipv4/ip_local_port_range", "File to read numbers from.")
    	flag.Parse()
    
    	file, err := os.Open(*filename)
    	defer file.Close()
    	if err != nil {
    		fmt.Printf("Failed to read file %q: %v\n", filename, err)
    	}
    	var start, end int
    	if n, err := fmt.Fscanf(file, "%d\t%d", &start, &end); n != 2 || err != nil {
    		if err == nil {
    			err = fmt.Errorf("unexpected count of parsed numbers (%d)\n", n)
    		}
    		fmt.Printf("Failed to parse port range from %s: %v\n", *filename, err)
    	} else {
    		fmt.Printf("Parsed range: %d - %d\n", start, end)
    	}
    }